src/reflect/scala/reflect/internal/Reporting.scala

```diff
@@ -116,6 +116,42 @@ abstract class Reporter {
     }
 }
 
+/** A `Reporter` that filters events before delegating.
+ *
+ *  Concrete subclasses should implement just the abstract `filter` method.
+ */
+abstract class DelegatingReporter(delegate: Reporter) extends Reporter {
```

Comment:
I think keeping this unused copy of `DelegatingReporter` around is more confusing than helpful - instead a comment in `reporters.DelegatingReporter` that the class should move together with `Reporter`?

Reply:
I started to transition `Reporter` but it's nontrivial. I didn't consider just making this a mix-in. I'll try that.